packages/report-flaky-tests/lib/run.ts

```diff
@@ -1,148 +1,116 @@
 /**
  * External dependencies
  */
-const path = require( 'path' );
-const github = require( '@actions/github' );
-const core = require( '@actions/core' );
-const unzipper = require( 'unzipper' );
-const { formatResultsErrors } = require( 'jest-message-util' );
+import * as github from '@actions/github';
+import * as core from '@actions/core';
 
-const TEST_RESULTS_LIST = {
-	open: `<!-- __TEST_RESULTS_LIST__ -->`,
-	close: `<!-- /__TEST_RESULTS_LIST__ -->`,
-};
-const TEST_RESULT = {
-	open: '<!-- __TEST_RESULT__ -->',
-	close: '<!-- /__TEST_RESULT__ -->',
-};
-
-const metaData = {
-	render: ( json ) => `<!-- __META_DATA__:${ JSON.stringify( json ) } -->`,
-	get: ( str ) => {
-		const matched = str.match( /<!-- __META_DATA__:(.*) -->/ );
-		if ( matched ) {
-			return JSON.parse( matched[ 1 ] );
-		}
-	},
-};
+/**
+ * Internal dependencies
+ */
+import { GitHubAPI } from './github-api';
+import {
+	renderIssueBody,
+	formatTestErrorMessage,
+	formatTestResults,
+	parseIssueBody,
+} from './markdown';
 
 async function run() {
 	const token = core.getInput( 'repo-token', { required: true } );
-	const label = core.getInput( 'label', { required: true } );
 	const artifactNamePrefix = core.getInput( 'artifact-name-prefix', {
 		required: true,
 	} );
 
-	const octokit = github.getOctokit( token );
+	const api = new GitHubAPI( token );
 
-	const flakyTests = await downloadReportFromArtifact(
-		octokit,
+	const flakyTests = await api.downloadReportFromArtifact(
 		artifactNamePrefix
 	);
 
 	if ( ! flakyTests ) {
+		// No flaky tests reported in this run.
 		return;
 	}
 
-	const issues = await fetchAllIssuesLabeledFlaky( octokit, label );
+	const label = core.getInput( 'label', { required: true } );
+	const issues = await api.fetchAllIssuesLabeledFlaky( label );
 
 	for ( const flakyTest of flakyTests ) {
-		const {
-			runner: testRunner = 'jest-circus',
-			title: testTitle,
-			path: testPath,
-			results: testResults,
-		} = flakyTest;
+		const { title: testTitle } = flakyTest;
 		const issueTitle = getIssueTitle( testTitle );
 		const reportedIssue = issues.find(
 			( issue ) => issue.title === issueTitle
 		);
-		const isTrunk = getHeadBranch() === 'trunk';
+		const testPath = flakyTest.path.startsWith( process.cwd() )
+			? flakyTest.path.slice( process.cwd().length )
+			: flakyTest.path;
```

Comment:
What does this do?

Reply:
It simply removes the starting path of the test file:

```diff
-/home/runner/work/gutenberg/gutenberg/test/e2e/specs/editor/various/copy-cut-paste.spec.js
+/test/e2e/specs/editor/various/copy-cut-paste.spec.js
```